apis/v1beta1/gateway_types.go

```diff
@@ -473,16 +542,52 @@ type GatewayAddress struct {
 	Value string `json:"value"`
 }
 
+// GatewayStatusAddress describes an address that is bound to a Gateway.
+type GatewayStatusAddress struct {
+	// Type of the address.
+	//
+	// +optional
+	// +kubebuilder:default=IPAddress
+	Type *AddressType `json:"type,omitempty"`
+
+	// Value of the address. The validity of the values will depend
+	// on the type and support by the controller.
+	//
+	// Examples: `1.2.3.4`, `128::1`, `my-ip-address`.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Value string `json:"value"`
+
+	// Routability specifies the routable bounds of this address
+	// Predefined values are: 'Private', 'Public', Cluster
+	// Other values MUST have a vendor prefix.
+	//
+	// Implementations that support Routability MUST populate this field
+	//
+	// +optional
+	// <gateway:experimental>
+	Routability *GatewayRoutability `json:"routability,omitempty"`
+}
+
 // GatewayStatus defines the observed state of Gateway.
 type GatewayStatus struct {
 	// Addresses lists the IP addresses that have actually been
 	// bound to the Gateway. These addresses may differ from the
 	// addresses in the Spec, e.g. if the Gateway automatically
 	// assigns an address from a reserved pool.
 	//
+	// Implementations that support GatewayRoutability MUST include an address
+	// that has the same routable semantics as defined in the Gateway spec.
+	//
+	// Implementations MAY add additional addresses in status, but they MUST be
+	// semantically less than the scope of the requested scope. For example if a
+	// user requests a `Private` routable Gateway then an additional address MAY
+	// have a routability of `Cluster` but MUST NOT include `Private`.
+	//
 	// +optional
 	// +kubebuilder:validation:MaxItems=16
-	Addresses []GatewayAddress `json:"addresses,omitempty"`
+	Addresses []GatewayStatusAddress `json:"addresses,omitempty"`
```

Comment:
Isn't this considered a breaking change?  Why not add a new field for `RoutableAddresses`?

Reply:
The guidance I had is that go types do not need to be backwards compatible as long as the yaml format is.